roles/convert2rhel/tasks/repositories.yml

```
@@ -0,0 +1,70 @@
+---
+- name: "Create credentials for 'Convert2RHEL' repository"
+  theforeman.foreman.content_credential:
+    username: "{{ foreman_username }}"
+    password: "{{ foreman_password }}"
+    server_url: "{{ foreman_server_url }}"
+    validate_certs: "{{ foreman_validate_certs | default(omit) }}"
+    organization: "{{ foreman_organization }}"
+    name: "Convert2RHEL CA"
+    content_type: cert
+    content: "{{ lookup('file', 'cdn-redhat-com-chain.pem') }}"
+    state: present
+
+- name: "Create 'Convert2RHEL 7' repository"
+  theforeman.foreman.repository:
+    username: "{{ foreman_username }}"
+    password: "{{ foreman_password }}"
+    server_url: "{{ foreman_server_url }}"
+    validate_certs: "{{ foreman_validate_certs | default(omit) }}"
+    organization: "{{ foreman_organization }}"
+    name: "{{ convert2rhel_rhel7_repo }}"
+    content_type: "yum"
+    product: "{{ convert2rhel_rhel7_product }}"
+    url: "https://cdn.redhat.com/content/public/convert2rhel/7/x86_64/os/"
+    # mirror_on_sync: true
+    verify_ssl_on_sync: true
+    download_policy: immediate
+    ssl_ca_cert: "Convert2RHEL CA"
+    state: present
+  when: foreman_content_rhel_enable_rhel7
+
+- name: "Create 'Convert2RHEL 8 'repository"
+  theforeman.foreman.repository:
+    username: "{{ foreman_username }}"
+    password: "{{ foreman_password }}"
+    server_url: "{{ foreman_server_url }}"
+    validate_certs: "{{ foreman_validate_certs | default(omit) }}"
+    organization: "{{ foreman_organization }}"
+    name: "{{ convert2rhel_rhel8_repo }}"
+    content_type: "yum"
+    product: "{{ convert2rhel_rhel8_product }}"
+    url: "https://cdn.redhat.com/content/public/convert2rhel/8/x86_64/os/"
+    # mirror_on_sync: true
+    download_policy: immediate
+    verify_ssl_on_sync: true
+    ssl_ca_cert: "Convert2RHEL CA"
+    state: present
+  when: foreman_content_rhel_enable_rhel8
+
+- name: "Synchronize 'Convert2RHEL 7' repository"
+  theforeman.foreman.repository_sync:
+    username: "{{ foreman_username }}"
+    password: "{{ foreman_password }}"
+    server_url: "{{ foreman_server_url }}"
+    validate_certs: "{{ foreman_validate_certs | default(omit) }}"
+    organization: "{{ foreman_organization }}"
+    repository: "{{ convert2rhel_rhel7_repo }}"
+    product: "{{ convert2rhel_rhel7_product }}"
+  when: foreman_content_rhel_enable_rhel7
```

the `content_rhel` role has a `foreman_content_rhel_sync_now` variable, to allow the user to decide whether they want a sync, or not.

I *think* we also should honor this here. @wbclark what do you think?

```suggestion
  when: foreman_content_rhel_enable_rhel7 and foreman_content_rhel_sync_now
```

---

So you mean to set `foreman_content_rhel_sync_now` on `convert2rhel` role too? Or can I get value of `foreman_content_rhel_sync_now` from `content_rhel` role?

---

Just note: the sync takes few seconds, repo is really small

---

I think (not tested) that if you `include_role`, the variables of that role are still available later in the play, so if the `include_role` remains (see my other comment), you shouldn't need to do anything

But yeah, if the repo is tiny, this might be over-optimization, I agree